drivers/sensor/lis2dh/lis2dh.c

```diff
@@ -321,6 +321,16 @@ int lis2dh_init(struct device *dev)
 		return status;
 	}
 
+#ifdef LIS2DH_HAS_SDO_SA0_PULL_UP
```

Comment:
Why not testing DT_INST_0_ST_LIS3DH directly?

Reply:
Now I also understand the previous commit... :D

Reply:
We can test LIS3DH directly. However, there are also other models which have the same "feature". One of them is already supported LIS2DH12. However there are no distinct bindings for it yet, so there is only check for LIS3DH for now.

Another reason to introducing `LIS2DH_HAS_SDO_SA0_PULL_UP` macro is to read the code in a more logical way... we disable SDO/SA0 pull-up only when the accelerometer has such feature.

Reply:
Seems legit. But take a look also to the possibility to have the pullup enable/disable in Kconfig (see below).

Reply:
@pabigot Thanks for review!
Basically you suggest that disconnecting should be fully controllable by user by specifying (or not) `disconnect-sdo-sa0-pull-up`. No automatic disconnect based on compatible string. I am ok with that. However in such case I do not like the idea of additional guarding based on compatible string. This would increase maintenance effort when new models would be added. Additionally I think we should trust the user when specyfying `disconnect-sdo-sa0-pull-up` property and just do what he requests us to do (disconnect pull-up).

Reply:
To be explicit: I'm fine with making it:
```
if (IS_ENABLED(DT_INST0_ST_LIS2DH_DISCONNECT_SDO_SDA0_PULL_UP)) {
   // code as below.
}
```
and letting anybody who sets that property on a device that doesn't support it deal with any side effects that might happen.

Reply:
@pabigot Just to note why there is no `IS_ENABLED()` in my commit, this is what gets generated when there is no `disconnect-sdo-sa0-pull-up` property specified:

```
#define DT_ST_STM32_SPI_FIFO_40013000_ST_LIS2DH_0_DISCONNECT_SDO_SA0_PULL_UP 0
#define DT_INST_0_ST_LIS2DH_DISCONNECT_SDO_SA0_PULL_UP DT_ST_STM32_SPI_FIFO_40013000_ST_LIS2DH_0_DISCONNECT_SDO_SA0_PULL_UP
```

Reply:
Yes, is that a problem?

Using `IS_ENABLED(M)` is standard for Kconfig and devicetree macros that have 0 or 1 as their value.

It also seems to incidentally fix the build problem with frdm_k64f on test_build_sensors_i_z where there is no define for any LIS2DH devicetree entries.  I think you should put it back.

Reply:
You are right, I have put it back. For some reason I had a feeling that IS_ENABLED was always true (also in case of 0). I have retested this case once again and it works as you wrote (IS_ENABLED is false in case there is no dts flag specified).